node/pkg/watchers/aptos/watcher.go

```diff
@@ -282,6 +321,87 @@ func (e *Watcher) retrievePayload(s string) ([]byte, error) {
 	return body, err
 }
 
+//nolint:noctx // TODO: this function should use a context.
+func (e *Watcher) queryIndexer(query string, headers map[string]string) ([]byte, error) {
+	// Create GraphQL request body
+	requestBody := map[string]interface{}{
+		"query": query,
+	}
+	jsonBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", e.aptosIndexerRPC, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return nil, err
+	}
+
+	// Set content type for GraphQL
+	req.Header.Set("Content-Type", "application/json")
+
+	// Add additional headers if provided
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	body, err := common.SafeRead(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, err
+}
+
+// processTransactionVersion fetches a transaction by version and extracts WormholeMessage events
+func (e *Watcher) processTransactionVersion(
+	logger *zap.Logger,
+	versionNum uint64,
+	sequenceForObserve uint64,
```

Comment:
if a transaction contains multiple wormhole messages, this will end up hitting the endpoint multiple times. do we need / is it worthwhile to specify the sequence here?

Reply:
Fixed